### PR TITLE
Use separate file for MARGINS in UxTheme

### PR DESCRIPTION
**Pull request by gpetrou**
## Proposed changes

- Move MARGINS struct to a separate file in UxTheme.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/winforms/pull/2217)

### CODE REVIEW

File: src/Common/src/SafeNativeMethods.cs [hunk 1]
```diff
@@ -223,7 +223,7 @@ internal static class SafeNativeMethods
         public static extern int GetThemePosition(HandleRef hTheme, int iPartId, int iStateId, int iPropId, out Point pPoint);
 
         [DllImport(ExternDll.Uxtheme, CharSet = CharSet.Auto)]
-        public static extern int GetThemeMargins(HandleRef hTheme, HandleRef hDC, int iPartId, int iStateId, int iPropId, NativeMethods.COMRECT prc, ref NativeMethods.MARGINS margins);
+        public static extern int GetThemeMargins(HandleRef hTheme, HandleRef hDC, int iPartId, int iStateId, int iPropId, NativeMethods.COMRECT prc, ref UxTheme.MARGINS margins);
```
Comment (JeremyKuhne) [inline src/Common/src/SafeNativeMethods.cs]: We don't have the static using in this class, you need to fully qualify `UxTheme`.

File: src/System.Windows.Forms/src/System/Windows/Forms/VisualStyles/VisualStyleRenderer.cs [hunk 2]
```diff
@@ -829,7 +829,7 @@ public Padding GetMargins(IDeviceContext dc, MarginProperty prop)
                 throw new InvalidEnumArgumentException(nameof(prop), (int)prop, typeof(MarginProperty));
             }
 
-            NativeMethods.MARGINS margins = new NativeMethods.MARGINS();
+            UxTheme.MARGINS margins = new UxTheme.MARGINS();
```
Comment (hughbe) [inline src/System.Windows.Forms/src/System/Windows/Forms/VisualStyles/VisualStyleRenderer.cs]: nit: use var in future

Comment (JeremyKuhne) [reply src/System.Windows.Forms/src/System/Windows/Forms/VisualStyles/VisualStyleRenderer.cs]: Better yet `UxTheme.MARGINS margins = default`. :)

File: src/Common/src/SafeNativeMethods.cs (hunk 1 above)
Comment (gpetrou) [reply src/Common/src/SafeNativeMethods.cs]: This builds fine for me locally and I can see the "using static Interop;". Or you mean something else?

File: src/System.Windows.Forms/src/System/Windows/Forms/VisualStyles/VisualStyleRenderer.cs (hunk 2 above)
Comment (gpetrou) [reply src/System.Windows.Forms/src/System/Windows/Forms/VisualStyles/VisualStyleRenderer.cs]: Sure.

File: src/Common/src/SafeNativeMethods.cs (hunk 1 above)
Comment (RussKie) [reply src/Common/src/SafeNativeMethods.cs]: I think we should have `using static Interop.UxTheme;` and then here `...prc, ref MARGINS margins);`

**Comment by RussKie** [comment]
The build is failing because the file file isn't referenced in System.Windows.Forms.Design.csproj:
```
D:\a\1\s\src\Common\src\SafeNativeMethods.cs(223,154): error CS0246: The type or namespace name 'UxTheme' could not be found (are you missing a using directive or an assembly reference?) [D:\a\1\s\src\System.Windows.Forms.Design\src\System.Windows.Forms.Design.csproj]
##[error]src\Common\src\SafeNativeMethods.cs(223,154): error CS0246: The type or namespace name 'UxTheme' could not be found (are you missing a using directive or an assembly reference?)
```

File: src/Common/src/SafeNativeMethods.cs [hunk 3]
```diff
@@ -187,7 +186,7 @@ internal static class SafeNativeMethods
...
```
Comment (RussKie) [inline src/Common/src/SafeNativeMethods.cs]:
```suggestion
        public static extern int GetThemeMargins(HandleRef hTheme, HandleRef hDC, int iPartId, int iStateId, int iPropId, NativeMethods.COMRECT prc, ref MARGINS margins);
```

Comment (gpetrou) [reply src/Common/src/SafeNativeMethods.cs]: Done.

**Comment by RussKie** [comment]
Thank you

**Pull request merged**